### PR TITLE
Creating sigmoid_shift.py under training/extensions

### PR DESCRIPTION
**Pull request by hsezhiyan**
Adding sigmoid shift in chainer/training/extensions.

This extension reduces a specified attribute of training (learning rate, alpha/beta values, etc.) using the following formula: e^{-1 * rate * i} / (e^{-1 * rate * i} + 1), where `i` is the iteration number.

Based on this formula, the specified attribute will approach 0 as training proceeds.

### CODE REVIEW

File: chainer/training/extensions/sigmoid_shift.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+
+class SigmoidShift(extension.Extension):
+
+    """Trainer extension to sigmoid shift an optimizer attribute.
+
+    This extension decreases the specified attribute of
+    an optimizer using the sigmoid function. This extension will
+    set the corresponding attribute to:
+    ``init_value * e^{-1 * rate * i} / (e^{-1 * rate * i} + 1)``,
```
Comment (toslunar) [inline chainer/training/extensions/sigmoid_shift.py]: Could you use `:math:` (inline) or `.. math::` (display)?

File: chainer/training/extensions/sigmoid_shift.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
+from __future__ import division
```
Comment (toslunar) [inline chainer/training/extensions/sigmoid_shift.py]: Is the line required?

File: chainer/training/extensions/sigmoid_shift.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        rate (float): Rate of sigmoid shift.
+        init (float): Initial value of the attribute. If it is ``None``, the
+            extension extracts the attribute at the first call and uses it as
+            the initial value.
+        target (float): Target value of the attribute. If the attribute reaches
+            this value, the shift stops.
+        optimizer (~chainer.Optimizer): Target optimizer to adjust the
+            attribute. If it is ``None``, the main optimizer of the updater is
+            used.
+    """
```
Comment (toslunar) [inline chainer/training/extensions/sigmoid_shift.py]: Could you cite a reference for the method?

File: chainer/training/extensions/sigmoid_shift.py (hunk 1 above)
Comment (toslunar) [reply chainer/training/extensions/sigmoid_shift.py]: OK to keep the formula in Python code as the other `*Shift` extensions do so.

File: chainer/training/extensions/sigmoid_shift.py (hunk 3 above)
Comment (hsezhiyan) [reply chainer/training/extensions/sigmoid_shift.py]: I don't believe this method is used specifically in any papers, and is therefore experimental.

**Comment by toslunar** [comment]
> I don't believe this method is used specifically in any papers, and is therefore experimental.

Because the method seems not enough recognized to increase the maintenance cost, I'm negative to adding the extension to `chainer.training.extensions`.

Nevertheless it may be a good opportunity to replace the guide [Trainer Extensions - Write a class inherited from the Extension class](https://docs.chainer.org/en/latest/guides/extensions.html#write-a-class-inherited-from-the-extension-class) with this `SigmoidShift`.  Please see #4800.

**Comment by Crissman** [comment]
@hsezhiyan Closing this PR due to the experimental nature of SigmoidShift. If you'd like to move this extension to the Trainer Extensions, please let me know the PR #, and we'll proceed.

Thanks!

**Pull request closed without merging**